### ISSUE TITLE
Add support for serializing/deserializing as an int instead of an object

### ISSUE THREAD

**Issue by autarch** [bitflags/bitflags#233]
As part of a project I'm working on, I wanted to match the representation of a C enum in my Rust code, so I used the `bitflags` crate. The C code generates a JSON object representing a parse tree. Some of the object values in this JSON object represent certain C enums, and they are simply integers, rather than a JSON object like `{"bits": 42}`. This means that attempting to deserialize these flags into a `bitflags`-created struct fails.

I wrote a crate of my own to do this, which you can see at https://github.com/houseabsolute/pg-pretty/tree/master/bitflags_serde_int.

I think it would make sense to just include this in the `bitflags` crate itself. If you agree, I'd be happy to work on a PR for this.

Do you have any suggestions for a better name for these macros? And if you don't want to include this, do you have package name suggestions? I'm not sure if it should be something like "serde_bitfflags", "bitflags_serde", or something else entirely.

**Comment by KodrAus** [comment]
I'd be happy to consider `serde` support for `bitflags`. For human-centric formats we could default to serializing a `|`-separated string of flags, and for machine-centric formats we could default to ints. It would then be possible for you to use `#[serde(with)]` to override that to always use `int`s in your project.

What do you think?

**Comment by autarch** [comment]
That sounds good. I'll look into this.

**Comment by autarch** [comment]
I started digging into this but I realized that this can't actually be part of the current crate without getting rid of the existing default serialization. If we want to add attribute macros for this like `#[derive(Serialize_bitflags, Deserialize_bitflags)]`, then the crate needs to export procedural macros. But a crate cannot export procedural macros and other types of code (macros, modules, etc.).

So either we'd have to change the existing serialization or make a new crate. I don't think this is worth breaking backcompat over. So would it make sense for me to just release a `serde_bitflags` crate with my code in it? I could add support for the human-readable format you suggested easily enough.
